### PR TITLE
Code quality issues - Remove unused code

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Cleanup

**What does this PR do? Why is it needed?**

Removes unused code found by DeepSource tool: https://deepsource.io/gh/rkapka/prysm/issue/SCC-U1000/occurrences/

**Which issues(s) does this PR fix?**

Part of #6747

**Other notes for review**

I did not remove unused code from `/validator/accounts/v2` because I didn't want to delete something useful for the ongoing wallet development. @rauljordan Is it OK to clean up these things?

### CODE REVIEW

File: beacon-chain/blockchain/metrics.go [hunk 1]
```diff
@@ -204,18 +193,6 @@ func reportEpochMetrics(state *stateTrie.BeaconState) {
 	}
 }
 
-// This captures metrics for block sent time by subtracts slot start time.
-func captureSentTimeMetric(genesisTime uint64, currentSlot uint64) error {
```
Comment (nisdas) [inline beacon-chain/blockchain/metrics.go:208]: this is pretty useful and I know we have use this for debugging  so not sure if we should remove it.
cc @terencechain

File: beacon-chain/db/kv/kv.go [hunk 2]
```diff
@@ -40,8 +39,6 @@ type Store struct {
 	databasePath        string
 	blockCache          *ristretto.Cache
 	validatorIndexCache *ristretto.Cache
-	stateSlotBitLock    sync.Mutex
```
Comment (nisdas) [inline beacon-chain/db/kv/kv.go:43]: @terencechain do we still need this ?

File: beacon-chain/operations/attestations/metrics.go [hunk 3]
```diff
@@ -31,8 +19,3 @@ var (
 		Help: "The number of expired and deleted block attestations in the pool.",
 	})
 )
-
-func (s *Service) updateMetrics() {
```
Comment (nisdas) [inline beacon-chain/operations/attestations/metrics.go:35]: this is pretty useful, I am inclined to keep this. I would like to know the count through our metrics

File: beacon-chain/p2p/network-keys [hunk 4]
```diff
@@ -0,0 +1 @@
+d0ec79110de61c63ee59a41006a8381c4a83f598fd9e3af072fdf94faa6b8d01
```
Comment (nisdas) [inline beacon-chain/p2p/network-keys:1]: remove

File: beacon-chain/state/stategen/replay.go [hunk 5]
```diff
@@ -318,34 +318,6 @@ func (s *State) archivedState(ctx context.Context, slot uint64) (*state.BeaconSt
 	return st, err
 }
 
-// This recomputes a state given the block root.
-func (s *State) recoverStateByRoot(ctx context.Context, root [32]byte) (*state.BeaconState, error) {
```
Comment (nisdas) [inline beacon-chain/state/stategen/replay.go:322]: @terencechain do we still want to keep this ?

File: beacon-chain/db/kv/kv.go (hunk 2 above)
Comment (terencechain) [reply beacon-chain/db/kv/kv.go:43]: nope, deprecated after #6461 

thanks for checking

File: beacon-chain/operations/attestations/metrics.go (hunk 3 above)
Comment (terencechain) [reply beacon-chain/operations/attestations/metrics.go:35]: agreed

File: beacon-chain/state/stategen/replay.go (hunk 5 above)
Comment (terencechain) [reply beacon-chain/state/stategen/replay.go:322]: no longer needed after #6461

File: beacon-chain/blockchain/metrics.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/blockchain/metrics.go:208]: I think it's ok to leave this on until beta. 

We plan to do a fine comb of metrics then.

File: validator/keymanager/v1/remote.go [hunk 6]
```diff
@@ -26,10 +26,9 @@ const (
 
 // Remote is a key manager that accesses a remote wallet daemon.
 type Remote struct {
-	paths               []string
-	conn                *grpc.ClientConn
-	accounts            map[[48]byte]*accountInfo
-	signClientInitiator func(*grpc.ClientConn)
+	paths    []string
+	conn     *grpc.ClientConn
+	accounts map[[48]byte]*accountInfo
```
Comment (nisdas) [inline validator/keymanager/v1/remote.go:31]: lets just leave it here and as it is

**Pull request merged**